app/services/reports/agency_invoice_supplement_report.rb

```diff
@@ -0,0 +1,22 @@
+module Reports
+  class AgencyInvoiceSupplementReport < BaseReport
+    REPORT_NAME = 'agency-invoice-supplemement-report'.freeze
+
+    def call
+      iaa_start_ends = ServiceProvider.distinct.
+        where.not(iaa: nil).
+        pluck(:iaa, :iaa_start_date, :iaa_end_date)
+
+      results = iaa_start_ends.flat_map do |iaa, iaa_start_date, iaa_end_date|
+        transaction_with_timeout do
+          Db::SpCost::SpCostSummaryByIaa.call(
+            iaa: iaa,
+            iaa_range: iaa_start_date..iaa_end_date,
+          )
+        end
+      end
```

Comment:
I moved the loop here because that's where the timeout method lives, and it seemed easier to have a series of transactions rather than one giant one?

Reply:
yes great idea!

Reply:
As long as the queries don't need to work from the same set of data, it should be fine? If data gets added or changed between transactions that changes the result of the report, we'd want to keep it as one transaction.